### DISCUSSION TITLE
code java pour commander des équipement(onvaluechanged)

### DISCUSSION THREAD

**Discussion by edouar1** [witsa/synapps#732] [Q&A]
bonjour; 
j'ai crée ce code sur l'évènement onvaluechanged
et isOutput = null; // Supposons que output contient l'état de votre lumière (true si allumée, false sinon)

// Fonction pour définir la visibilité des icônes en fonction de l'état de la lumière
const setVisible = (elementID, isVisible) => {
    const element = document.getElementById(elementID);
    if (element) {
        element.style.display = isVisible ? "block" : "none";
    }
};

// Supposons que vous avez les IDs des icônes HTML pour la lumière éteinte et allumée
const colorNOKIconID = "colorNOK";
const colorOKIconID = "colorOK";

// Supposons que vous avez une variable output qui représente l'état de la lumière (true si allumée, false sinon)
if (isOutput === null) {
    console.error("Erreur : aucune donnée n'a été reçue.");
} else {
    // Mettre à jour l'affichage des icônes en fonction de l'état de la lumière
    switch (isOutput) {
        case true: // Lumière allumée
            setVisible(colorNOKIconID, false);
            setVisible(colorOKIconID, true);
            break;
        case false: // Lumière éteinte
            setVisible(colorNOKIconID, true);
            setVisible(colorOKIconID, false);
            break;
        default:
            console.error("Erreur : état de lumière invalide.");
            break;
    }
}
mais ca ne marche pas .
voila comment je les relier 
![image](https://github.com/witsa/synapps/assets/170326568/684e8ff8-762d-43b0-b110-cc04ddabc98e)
Ducoup l'évènement est ce qu'il sera sur les couleurs ou les icône ou bien un seul événement suffit de contrôler la lampe?
merci pour votre aide .
cordialement.

**Comment by manufitoussiwit** [comment]
Bonjour,

Pour vous aidez, je peux vous donner quelques éléments : 

```js

// l'acteur en cours : this ou bien context.target 
// par exemple : 
// changer la propriété couleur de l'acteur
this.properties.color = 'red';

// ou bien 
context.target.properties.color = 'red';

// pour aller chercher un acteur et le placer dans variable actor:
const actor = context.getActor('<la clé de l acteur>');

// pour aller chercher la scène ou le composite dans lequel est l'acteur :

context.stage;
// ou bien
this.stage;

// pour changer une additionnelle :

this.additionals.prop1 = 'salut!';

// dans les scripts onValueChanged et onPropertyValueChanged, on a accès à : 
// la nouvelle value : 
context.newValue;

// l'ancienne : 
context.oldValue;

// dans les scripts onPropertyValueChanged uniquement : 
// on a accès à le chemin de propriété ou additionnelle qui a changé : 
context.propertyPath;

// par exemple ça peut être la valeur ; 'properties.color' ou bien 'additionals.prop3'.

// dans un script de transformation de liaison comme onReadTransform
// la valeur reçue : 
context.value;

// c'est une transformation, donc il faut retourner la valeur à propriété (d'ou le return context.value par défaut)



```

Pour connaitre comment accéder par script à une additionnelle ou une propriété, on peut copier son chemin. [voir ici](https://witsa.github.io/synapps/synapps-studio-releases/notes/1.2.0#copie-de-chemin-de-propri%C3%A9t%C3%A9additionnelle)

Pour comprendre les évènements et le cycle de vie des acteurs, je recommande de consulter ceci : https://witsa.github.io/synapps/concepts/scripts/actor-life-cycle.html

Dans le cycle de vie par exemple, il faut savoir qu'on peut avoir accès aux éléments HTML du DOM uniquement à partir de l'évènement onPostInit.

En espérant que sa vous aide un peu.
cordialement

**Comment by edouar1** [comment]
merci pour votre aide.
cdt.

**Comment by edouar1** [comment]
Bonjour, 
j'ai une question je veut savoir c'est quoi la différence entre listes reflet et listes variable , fournisseurs de reflet et variable
merci 
cordialement
